src/wsdiscoveryclient.cpp

```diff
@@ -73,15 +75,20 @@ void WSDiscoveryClient::sendProbe(const QList<KDQName> &typeList, const QList<QU
     KDSoapMessageAddressingProperties addressing;
     addressing.setAddressingNamespace(KDSoapMessageAddressingProperties::Addressing200408);
     addressing.setAction(QStringLiteral("http://schemas.xmlsoap.org/ws/2005/04/discovery/Probe"));
-    addressing.setMessageID(QStringLiteral("urn:uuid:") + QUuid::createUuid().toString(QUuid::WithoutBraces));
+    addressing.setMessageID(QStringLiteral("urn:uuid:") + QUuid::createUuid().toString().remove("{").remove("}"));
```

Comment:
Yuck, not very nice. But I understand you want Qt 5.9 compatibility. I suggest make the compatibility more explicit like this:
```suggestion
    #if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
        addressing.setMessageID(QStringLiteral("urn:uuid:") + QUuid::createUuid().toString(QUuid::WithoutBraces));
    #else
        addressing.setMessageID(QStringLiteral("urn:uuid:") + QUuid::createUuid().toString().remove("{").remove("}"));
    #endif
```
This way we can delete the compatibility code once Qt 5.9 is not supported anymore.

Reply:
Looks reasonable.